mypyc/genops.py

```diff
@@ -1307,59 +1322,73 @@ def translate_call(self, expr: CallExpr, callee: Expression) -> Value:
+        # they check that everything in arg_kinds is ARG_POS.
 
-        fullname = callee.fullname
-        if fullname == 'builtins.len' and len(expr.args) == 1 and expr.arg_kinds == [ARG_POS]:
-            expr_rtype = args[0].type
+        # Special case builtins.len
+        if (callee.fullname == 'builtins.len'
+                and len(expr.args) == 1
+                and expr.arg_kinds == [ARG_POS]):
+            expr_rtype = arg_values[0].type
             if isinstance(expr_rtype, RTuple):
                 # len() of fixed-length tuple can be trivially determined statically.
                 return self.add(LoadInt(len(expr_rtype.types)))
-        if (fullname == 'builtins.isinstance'
+
+        # Special case builtins.isinstance
+        if (callee.fullname == 'builtins.isinstance'
                 and len(expr.args) == 2
                 and expr.arg_kinds == [ARG_POS, ARG_POS]
                 and isinstance(expr.args[1], RefExpr)
                 and isinstance(expr.args[1].node, TypeInfo)
                 and self.is_native_module_ref_expr(expr.args[1])):
             # Special case native isinstance() checks as this makes them much faster.
-            return self.primitive_op(fast_isinstance_op, args, expr.line)
+            return self.primitive_op(fast_isinstance_op, arg_values, expr.line)
 
         # Handle data-driven special-cased primitive call ops.
-        if fullname is not None and expr.arg_kinds == [ARG_POS] * len(args):
-            ops = func_ops.get(fullname, [])
-            target = self.matching_primitive_op(ops, args, expr.line)
+        if callee.fullname is not None and expr.arg_kinds == [ARG_POS] * len(arg_values):
+            ops = func_ops.get(callee.fullname, [])
+            target = self.matching_primitive_op(ops, arg_values, expr.line)
             if target:
                 return target
 
-        fn = callee.fullname
-        # Try to generate a native call.
-        if signature and fn:
-            # Native call
+        # Don't rely on the inferred callee type, since it may have type
+        # variable substitutions that aren't valid at runtime (due to type
+        # erasure). Instead pick the declared signature of the native function
+        # as the true signature.
+        signature = self.get_native_signature(callee)
+
+        # Standard native call if signature and fullname are good and all arguments are positional
+        # or named.
+        if (signature is not None
+                and callee.fullname is not None
+                and all(kind in (ARG_POS, ARG_NAMED) for kind in expr.arg_kinds)):
+            # Normalize keyword args to positionals.
+            arg_values_with_nones = self.keyword_args_to_positional(arg_values,
+                                                                    expr.arg_kinds,
+                                                                    expr.arg_names,
+                                                                    signature)
```

Comment:
Formatting suggestion: break the line after the `(` and keep the args on one line